honeybee_3dm/grid.py

```diff
@@ -27,43 +28,48 @@ def import_grids(rhino3dm_file, tolerance, grid_size=1, grid_offset=0):
         tolerance: A rhino3dm tolerance object. Tolerance set in the rhino file.
         grid_size: Grid size. Defaults to 1.
         grid_offset: Grid offset from the grid geometry. Defaults to 0.
-
+        visibility: Bool. If set to False then the objects on an "off"
+            layer in Rhino3dm will also be imported. Defaults to True.
+        config: A dictionary. The config file as a dictionary object.
+            Defaults to None.
     Returns:
         A list of Honeybee grids.
     """
-
-    # Objects from rhino file
-    try:
-        grid_objs = filter_objects_by_layer(rhino3dm_file, 'grid')
-        warnings.warn(
-            'Only objects on layer "grid" will be used to create grids.')
-    except ValueError:
-        # no layer named grid
-        return []
-
     hb_grids = []
+    config = config
+    # If config file is provided and any Rhino layer is tied to HB_grid layer
+    if config and config['HB_layers']['HB_grid']:
+        grid_objs = objects_on_parent_child(rhino3dm_file, 
+            config['HB_layers']['HB_grid'], visibility=visibility)
```

Comment:
You can avoid having several nested layers of code. It is really hard to follow the logic of 4-5 if/else one after another.

```python
if not config or 'HB_grid' not in config['HB_layers']:
   return hb_grids

# now start the new check here
```